src/plugins/vis_types/vega/public/vega_view/vega_base_view.styles.ts

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { css } from '@emotion/react';
+import {
+  UseEuiTheme,
+  logicalCSS,
+  mathWithUnits,
+  euiScrollBarStyles,
+  euiFontSize,
+  euiBackgroundColor,
+} from '@elastic/eui';
+
+// Note that these are globally-set styles, due to vega_base_view being vanilla JS/non-React
+export const vegaBaseViewStyles = (euiThemeContext: UseEuiTheme) => {
+  const { euiTheme } = euiThemeContext;
+  return css`
+    .vgaVis {
+      display: flex;
+      flex: 1 1 100%;
+      position: relative;
+      // flex-direction determined by js
+    }
+
+    .vgaVis--autoresize {
+      ${euiScrollBarStyles(euiThemeContext)};
```

Comment:
It does look like the scrolling behavior changed here, but it looks like an improvement to me

Before:
https://user-images.githubusercontent.com/315764/199773632-6888436d-d991-4ac8-94dd-3c0224daf8d2.mov

After:
https://user-images.githubusercontent.com/315764/199773672-f787630c-b8a6-4420-b549-4d7d4d146302.mov

Reply:
> In the original CSS, euiScrollbar mixin was applied in .vgaVis__view. I'm curious why it got moved to the .vgaVis--autoresize

Curious where you're getting that from? There's no scrollbar styles applied whatsoever to `.vgaVis__view` that I can see.

https://github.com/elastic/kibana/blob/f07f27e37b2ba1095880894b6679c48dc4619552/src/plugins/vis_types/vega/public/components/vega_vis.scss#L27-L32

vs

https://github.com/elastic/kibana/blob/f07f27e37b2ba1095880894b6679c48dc4619552/src/plugins/vis_types/vega/public/components/vega_vis.scss#L17-L18

It's been a few weeks and one major medical procedure since I converted that Vega code as an example so I'll confess I'm can't quite 100% remember what I changed/fixed that would cause the before/after. At a quick glance it's  It's possible that the CSS-in-JS mixins/vars being used are causing the improved scrollbar experience (e.g. updated sizes).

I can definitely look into the remaining bugs you included above but would super appreciate potentially pairing on them to get my local in a state where I can diagnose the issues! I'll also mention that https://github.com/elastic/kibana/pull/141279/commits/319243bb6f026b5ff30e02660759453bec99e9ca is very much an optional commit - I was basically just curious to see what part of Kibana would look like on Emotion and off Sass - and if it's unintentionally wreaking havoc/causing bugs down the line I can absolutely revert it. The tooltips are the only major things required to be in CSS-in-JS, because 1. they're doing custom stuff, and 2. we've removed the global CSS classNames they were reliant on.

Reply:
Actually, sorry, after looking more closely: `Warning colors are gone on Vega` is likely the only screenshot related to https://github.com/elastic/kibana/commit/319243bb6f026b5ff30e02660759453bec99e9ca. The remaining issues look related to EuiButton's partial conversion to Emotion and may be relying on EUI classes that are now removed.

`Add layer button in dark mode got a lot brighter` does look intentional however, the EUI designers have been tweaking/increasing color contrast automatically along the way.